test/unit-tests/liquidations.test.ts

```diff
@@ -0,0 +1,442 @@
+    scaleNum(0.237, 27),
+    scaleNum(0.3326, 27),
+    scaleNum(0.4032, 27),
+    scaleNum(0.4603, 27),
+  ]
+
+  before('set up contracts', async () => {
+    // setup usdc and weth
+    usdc = await MockERC20.new('USDC', 'USDC', usdcDecimals)
+    dai = await MockERC20.new('DAI', 'DAI', daiDecimals)
+    weth = await MockERC20.new('WETH', 'WETH', wethDecimals)
+
+    const now = (await time.latest()).toNumber()
+    expiry = now + time.duration.days(30).toNumber()
+    // initiate addressbook first.
+    addressBook = await MockAddressBook.new()
+    // setup oracle
+    oracle = await MockOracle.new()
+    await addressBook.setOracle(oracle.address)
+    // setup calculator
+    calculator = await MarginCalculator.new(oracle.address, {from: owner})
+    // set collateral dust
+    await calculator.setCollateralDust(weth.address, wethDust, {from: owner})
+    await calculator.setCollateralDust(usdc.address, usdcDust, {from: owner})
+    // set product spot shock value
+    await calculator.setSpotShock(weth.address, usdc.address, usdc.address, true, productSpotShockValue)
+    await calculator.setSpotShock(weth.address, usdc.address, weth.address, false, productSpotShockValue)
+    // set time to expiry and each upper bound value
+    for (let i = 0; i < expiryToValue.length; i++) {
+      // set for put product
+      await calculator.setTimeToExpiryValue(
+        weth.address,
+        usdc.address,
+        usdc.address,
+        true,
+        timeToExpiry[i],
+        expiryToValue[i],
+        {from: owner},
+      )
+      await calculator.setProductTimeToExpiry(weth.address, usdc.address, usdc.address, true, timeToExpiry[i], {
+        from: owner,
+      })
+
+      // set for call product
+      await calculator.setTimeToExpiryValue(
+        weth.address,
+        usdc.address,
+        weth.address,
+        false,
+        timeToExpiry[i],
+        expiryToValue[i],
+        {from: owner},
+      )
+      await calculator.setProductTimeToExpiry(weth.address, usdc.address, weth.address, false, timeToExpiry[i], {
+        from: owner,
+      })
+    }
+  })
+
+  describe('check if vault is liquidatable', () => {
+    const oracleDeviation = 0.05
+
+    const strikePrice = 200
+    const scaledStrikePrice = createTokenAmount(strikePrice)
+    const isPut = true
+
+    let shortOtoken: MockOtokenInstance
+    let optionExpiry: BigNumber
+
+    beforeEach(async () => {
+      const oracleDeviationValue = scaleNum(oracleDeviation, 27)
+      await calculator.setOracleDeviation(oracleDeviationValue, {from: owner})
+
+      optionExpiry = new BigNumber(await time.latest()).plus(timeToExpiry[1])
+
+      shortOtoken = await MockOtoken.new()
+      await shortOtoken.init(
+        addressBook.address,
+        weth.address,
+        usdc.address,
+        usdc.address,
+        scaledStrikePrice,
+        optionExpiry,
+        isPut,
+      )
+    })
+
+    it('should not be able to liquidate vault with type equal to 1', async () => {
```

Comment:
it should be able to liquidate this right?

Reply:
Yup a typo mistake 😅